spec/lib/adviser_list_csv_spec.rb

```diff
@@ -1,13 +1,68 @@
 RSpec.describe AdviserListCsv do
   describe '#to_csv' do
+    let(:adviser1) { FactoryGirl.create(:adviser, adviser1_certifications) }
+    let(:adviser2) { FactoryGirl.create(:adviser, adviser2_certifications) }
+    let(:advisers) { [adviser1, adviser2] }
+
+    let(:qualifications) { 3.times.map { FactoryGirl.create(:qualification) } }
+    let(:accreditations) { 3.times.map { FactoryGirl.create(:accreditation) } }
+
+    let(:adviser1_certifications) do
+      {
+        qualifications: [qualifications.first, qualifications.last],
+        accreditations: [accreditations.first]
+      }
+    end
+    let(:adviser2_certifications) do
+      {
+        qualifications: [qualifications.last],
+        accreditations: [accreditations.last]
+      }
+    end
+
+    let(:subject)  { described_class.new(advisers) }
+
     it 'responds to csv conversion' do
-      expect(described_class.new []).to respond_to(:to_csv)
+      expect(subject).to respond_to(:to_csv)
+    end
+
+    context 'when csv is generated' do
+      let(:table)    { CSV.parse(subject.to_csv) }
+      let(:headings) { table[0] }
+      let(:row1)     { table[1] }
+      let(:row2)     { table[2] }
+
+      it 'returns csv headings' do
+        expect(headings).to eq(
+          [
+            'Ref. Number',
+            'Name',
+            'Firm'
+          ] + Qualification.pluck(:name) + Accreditation.pluck(:name)
+        )
+      end
+
+      it 'returns data for each adviser' do
+        expect(row1).to eq([
+          adviser1.reference_number,
+          adviser1.name,
+          adviser1.firm.registered_name,
+          'Y', 'N', 'Y', 'Y', 'N', 'N'
+        ])
+
+        expect(row2).to eq([
+          adviser2.reference_number,
+          adviser2.name,
+          adviser2.firm.registered_name,
+          'N', 'N', 'Y', 'N', 'N', 'Y'
+        ])
+      end
     end
-  end
 
-  context 'Office' do
-    it 'should not respond to csv conversion' do
-      expect(Office).to_not respond_to(:to_csv)
+    context 'Office' do
```

Comment:
This should be a `describe` rather than a `context`. The question is.. does `Office` ever respond to the `#csv` method? If it does then we need to add that context. If it doesn't then a `describe` which ascertains that neither the `Office` or any instance of the `Office` class responds to `csv`. 

I hope it makes sense. If doesn't we can have a quick call over Slack to discuss this.

Reply:
`describe` is used for methods `to_csv` is the method and in this case we are testing it with `Office` so feel like what we are doing here is fine?

Generally I feel this test is kinda strange (it was from the original code) and I would rather we remove it totally.

Reply:
@Tr4pSt3R perhaps I miss understood. This lines were already there and introduced here: https://github.com/moneyadviceservice/rad/commit/a50e1badca4e593f56c2008488423454639074bf#diff-59aaa67083461a3d95ed7e719c81ca06R8

Reply:
@tomas-stefano the lines were there but I'm saying it should look something like this

```ruby
   1 RSpec.describe AdviserListCsv do
   2   describe '#to_csv' do
   3     it 'responds to csv conversion' do
   4       expect(described_class.new []).to respond_to(:to_csv)
   5     end
   6
   7     context 'Office' do
   8       it 'should not respond to csv conversion' do
   9         expect(Office).to_not respond_to(:to_csv)
  10       end
  11     end
  12   end
  13 end
```
When you see it as written above then you'd see why having that context above is truly redundant.  This test for `Office` responding to the `to_csv` method should be scrapped.

Rather than, something like this...
```ruby
   1 RSpec.describe AdviserListCsv do
   2   describe '#to_csv' do
   3     it 'responds to csv conversion' do
   4       expect(described_class.new []).to respond_to(:to_csv)
   5     end
   6   end
   7
   8   context 'Office' do
   9     it 'should not respond to csv conversion' do
  10       expect(Office).to_not respond_to(:to_csv)
  11     end
  12   end
  13 end
```

At any rate this test that the Office class does not respond to `to_csv` is unnecessary.

Reply:
This two specs are unnecessary. Usually we use respond_to spec when we want to force some kind of interface, which is not the case.

Reply:
removed `respond_to` tests 👍